### ISSUE TITLE
Animals Classification

### ISSUE THREAD

**Issue by siddhant4ds** [abhisheks008/DL-Simplified#482]
### Deep Learning Simplified Repository (Proposing new issue)
:red_circle: **Project Title** : Animals Classification</br>
:red_circle: **Aim** :  Create a DL model which will classify 5 categories of animals: cats, dogs, elephants, horses & lions.</br>
:red_circle: **Dataset** : [Animals classification dataset](https://www.kaggle.com/datasets/antobenedetti/animals) </br>
:red_circle: **Approach** : Try to use 3-4 algorithms to implement the models and compare all the algorithms to find out the best fitted algorithm for the model by checking the accuracy scores. Also do not forget to do a exploratory data analysis before creating any model.

**********************************************************************************
### 📍 **Follow the Guidelines to Contribute in the Project :**
* You need to create a separate folder named as the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.

***********************************************************************
:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before start Contributing.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name : 
- GitHub Profile Link : 
- Email ID :
- Participant ID (if applicable):
- Approach for this Project :
- What is your participant role? (Mention the Open Source program)

*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

**Comment by siddhant4ds** [comment]
I would like to work on this issue.

* Full name : Siddhant Tiwari
* GitHub Profile Link : https://github.com/siddhant4ds
* Email ID : siddhant.tiwari.ds@gmail.com
* Participant ID (if applicable): N/A
* Approach for this Project : 
   1. Baseline model using a CNN.
   2. Data augmentation techniques for improving data quality.
   3. Transfer-learning using 5-6 architectures as feature-extraction backbones.
   4. Fine-tuning some of the best performing transfer learning models.

* What is your participant role? (Mention the Open Source program) SWOC-4 (2024)

**Comment by abhisheks008** [comment]
Assigned under SWOC @siddhant4ds

**Comment by Yuvika-14** [comment]
I would like to work on this issue.

Full name : Yuvika Singh

GitHub Profile Link : https:// github.com/YuvikaSingh

Email ID : yuvika1406042gmail.com



Participant ID (if applicable): N/A

Approach for this Project :

Baseline model using a CNN.
Data augmentation techniques for improving data quality.

Fine-tuning some of the best performing transfer learning models.

**Comment by abhisheks008** [comment]
Hi @Yuvika-14  wait for the induction session to complete by today evening, after that issues will be assigned to the contributors.

**Comment by Rithish5513U** [comment]
I would like to work on this issue.

Full name : Rithish S

GitHub Profile Link :https://github.com/Rithish5513U

Email ID : rithish.satish@gmail.com

Participant ID (if applicable): N/A

Approach for this Project :

1.Creating a foundational convolutional neural network (CNN) model.
2.Employing data augmentation methods to enhance data quality.
3.Adjusting and refining select top-performing transfer learning models.

**Comment by abhisheks008** [comment]
Hi @Rithish5513U and @Yuvika-14 can you guys please share your approach elaborately?

**Comment by chirag-garg9** [comment]
I have previous experiences related to such classification projects and will love to solve this issue.

Full name : Chirag Garg

GitHub Profile Link : https://github.com/chirag-garg9
Email ID : chirag.garg.5293@gmail.com

Participant ID (if applicable): N/A

Approach for this Project :

1.Data augmentation techniques for improving data quality and to make my model more robust.
2.Transfer-learning using 5-6 architectures as feature-extraction backbones such as Resnet,VGG16,etc.
3. Adding a custom top and classifying using the fully connected layers and classification layer at the end.
4. Fine-Tuning model the above data if required and selecting the best performing model using confusion matrix and other useful parameters.

Participant Role : GSSOC Contributor'24

**Comment by abhisheks008** [comment]
Well after considering the comments I prefer to assign this issue to @chirag-garg9. You guys can work on other open issues.

**Comment by siddhant4ds** [comment]
@abhisheks008 I had created this issue towards the end of SWOC-4 but ran out of time while working on it. I would like to complete that work and contribute this project to the repo as an independent contributor. Kindly assign this issue to me if it is available. Thanks.

**Comment by abhisheks008** [comment]
Assigned @siddhant4ds
